src/app/core/components/language-picker/language-picker.component.ts

```diff
@@ -13,15 +13,15 @@ export class LanguagePickerComponent implements OnChanges {
   @Output() changeLang = new EventEmitter<string>();
 
   readonly languages = this.localeService.languages;
-  current = this.languages.find(l => l.tag === this.localeService.currentTag);
+  current = this.localeService.currentLang;
 
   constructor(
     private localeService: LocaleService,
   ) {}
 
   ngOnChanges(): void {
     if (this.defaultLang) {
-      this.current = this.languages?.find(l => l.tag === this.defaultLang);
+      this.current = this.languages.find(l => l.tag === this.defaultLang);
```

Comment:
You both set `current` to the currentLang of locale service and set it to the default ?
This default to local current is strange when you set a value...
Maybe you could not set default value to current and in `ngOnChanges`:
```ts
this.current = this.defaultLang ? this.languages.find(l => l.tag === this.defaultLang) : this.localeService.currentLang;
```

Reply:
Fixed